mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -62,20 +60,25 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter, IReplaceableNode {
         }
 
     override fun replaceNode(concept: ConceptReference?): INode {
-        requireNotNull(concept) { "Can't replace $node with null concept. Use BaseConcept explicitly." }
+        requireNotNull(concept) { "Cannot replace node `$node` with a null concept. Explicitly specify a concept (e.g., `BaseConcept`)." }
+        val parent = requireNotNull(node.parent) { "Cannot replace node `$node` because it has no parent." }
+
+        val mpsConcept = MPSConcept.tryParseUID(concept.uid)
+        requireNotNull(mpsConcept) { "Concept UID `${concept.uid}` cannot be parsed as MPS concept." }
+        val sConcept = MetaAdapterByDeclaration.asInstanceConcept(mpsConcept.concept)
 
         val id = node.nodeId
         val model = checkNotNull(node.model) { "Node is not part of a model" }
-        val newNode = model.createNode(SConceptAdapterById(SConceptId.deserialize(concept.uid), ""), id)
+        val newNode = model.createNode(sConcept, id)
```

Comment:
`model` being null should be allowed and then we can use `SNode(id)` instead of `model.createNode`.

Reply:
Done